sound/soc/sof/ipc4-topology.c

```diff
@@ -497,7 +499,20 @@ static int sof_ipc4_widget_setup_msg(struct snd_sof_widget *swidget, struct sof_
 
 	msg->extension = SOF_IPC4_MOD_EXT_CORE_ID(swidget->core);
 
-	type = (fw_module->man4_module_entry.type & SOF_IPC4_MODULE_DP) ? 1 : 0;
+	switch (swidget->comp_domain) {
+	case SOF_COMP_DOMAIN_LL:
+		type = 0;
+		break;
+	case SOF_COMP_DOMAIN_DP:
+		type = 1;
+		break;
+	default:
```

Comment:
this is a bit odd that default is before the UNSET case.

Reply:
what I'd do is maybe start with the default domain based on the manifest and override it with the type from topology. That way you can simply break in the case of default and not even check the UNSET case.